namespace/bcl.adoc

```diff
@@ -0,0 +1,46 @@
+= Binary Compatibility Levels
+
+This is a classification of the levels of binary compatibility support for Java EE 8 classic applications in terms of the level of difficulty/complexity of the associated implementation and tools.
+
+== Level 0 - Jakarta EE 8 Only
+* No transformation required
+* Requires a Jakarta EE 8 container
+* No mandated support for backwards compatibility
+** Up to implementors to decide what level of support to provide
+
+== Level 1 - Jakarta EE 8 Translation to EE 9
+* Requires a direct translation of javax.* packages to jakarta.* analog
+* Requires transformation of META-INF/services to jakarta analogs
+* Requires transformation of properties files referencing javax based properties
+* Requires transformation of server specific properties that are classic javax based properties
+** This could be a specfic build or deployment step that processes an EE 8 app
+* Requires only a Jakarta EE 9 container
+* Requires the big bang javax to jakarta approach
+** This simplifies the EE9 container
+
+== Level 2 - Colocated Jakarta EE 8/9 with Isolation
+* Requires an isolated Jakarta EE 8 container
+* Requires an isolated  Jakarta EE 9 container
+* No interaction of EE 8 and EE 9 components
+* Requires the big bang javax to jakarta approach
```

Comment:
not necessarily.   The isolated EE9 can be whatever incremental or big bang EE spec it wants to be, because it is isolated.

Reply:
I agree.  The isolated jakarta.* runtime could grow over time.

Reply:
I was thinking that isolation of the Jakarta EE 9 container could be more problematic wtihout the big bang, but I guess it could be managed.